x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/heuristic/PValueScore.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+
+package org.elasticsearch.xpack.ml.aggs.heuristic;
+
+
+import org.apache.commons.math3.util.FastMath;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+import org.elasticsearch.search.aggregations.bucket.terms.heuristic.NXYSignificanceHeuristic;
+import org.elasticsearch.search.aggregations.bucket.terms.heuristic.SignificanceHeuristic;
+
+import java.io.IOException;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class PValueScore extends NXYSignificanceHeuristic {
+    public static final String NAME = "p_value";
+    public static final ConstructingObjectParser<PValueScore, Void> PARSER = new ConstructingObjectParser<>(NAME, args -> {
+        boolean backgroundIsSuperset = args[0] == null || (boolean) args[0];
+        return new PValueScore(backgroundIsSuperset);
+    });
+    static {
+        PARSER.declareBoolean(optionalConstructorArg(), BACKGROUND_IS_SUPERSET);
+    }
+
+    private static final MlChiSquaredDistribution CHI_SQUARED_DISTRIBUTION = new MlChiSquaredDistribution(1);
+
+    public PValueScore(boolean backgroundIsSuperset) {
+        super(true, backgroundIsSuperset);
+    }
+
+    public PValueScore(StreamInput in) throws IOException {
+        super(true, in.readBoolean());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(backgroundIsSuperset);
```

Comment:
I would be tempted to use the `super` methods for writeTo and in the `StreamInput` ctor in case something changes in the base class.  This class can't be constructed with `includeNegatives == false` so that constraint is preserved.